packages/jaeger-ui/src/utils/interval-calculator.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const time = new (function() {
+  this.second = 1000;
+  this.minute = 60 * this.second;
+  this.hour = 60 * this.minute;
+  this.day = 24 * this.hour;
+})();
+
+const intervalMin = [
+  [time.second * 1, time.second * 0.5], // < 1s
+  [time.second * 5, time.second * 1], // < 5s
+  [time.second * 10, time.second * 5], // < 10s
+  [time.second * 30, time.second * 10], // < 30s
+  [time.minute * 1, time.second * 30], // < 1m
+  [time.minute * 3, time.minute * 1], // < 3m
+  [time.minute * 5, time.minute * 2], // < 5m
+  [time.minute * 10, time.minute * 5], // < 10m
+  [time.minute * 40, time.minute * 10], // < 40m
+  [time.hour * 1, time.minute * 30], // < 1h
+  [time.hour * 2, time.hour * 1], // < 2h
+  [time.hour * 6, time.hour * 3], // < 6h
+];
+
+const roundInterval = interval => {
+  const defaultInterval = [time.minute * 10, time.minute * 5];
+
+  const chosenInterval = intervalMin.find(limit => interval < limit[0]);
+
+  return chosenInterval || defaultInterval;
+};
+
+const getIntervalMilliSecond = (timeRange, size) => roundInterval(timeRange / size)[1];
```

Comment:
please define as function, with data types

Reply:
I don't think `size` parameter is relevant to this function's purpose

Reply:
but this is the amount of data points, how can we decide what will be the step without it?

Reply:
the caller can do the division, this helper can be just about picking the closest even interval.

Reply:
@galangel @AlonMiz WDYT?

Reply:
i think that the return type is also poorly named
let's return an object

const getIntervalMilliSecond = (timeRange: type1, numOfExpectedPoints: number) => {
  return roundInterval(timeRange / size).interval
}